src/MatrixFields/field_name_dict.jl

```diff
@@ -181,11 +181,51 @@ function get_internal_entry(
         entry
     elseif name_pair[2] == @name() && broadcasted_has_field(T, name_pair[1])
         # multiplication case 2 or 4, second argument
-        Base.broadcasted(entry) do matrix_row
-            map(matrix_row) do matrix_row_entry
-                broadcasted_get_field(matrix_row_entry, name_pair[1])
-            end
-        end # Note: This assumes that the entry is in a FieldMatrixBroadcasted.
+        target_field_eltype = broadcasted_get_field_type(T, name_pair[1])
+        if target_field_eltype <: Number
```

Comment:
```suggestion
        if sizeof(target_field_eltype) == sizeof(eltype(parent(entry))
```
This could potentially be incorrect if we have a `Number` with multiple fields (e.g., `Rational` or `Dual`). What we really want to check here is whether the `target_field_eltype` contains exactly one value from the parent array of `entry`.